pkgs/desktops/budgie/budgie-control-center/default.nix

```diff
@@ -0,0 +1,134 @@
+{ lib
+, stdenv
+, substituteAll
+, fetchFromGitHub
+, accountsservice
+, budgie-desktop
+, clutter-gtk
+, clutter
+, colord-gtk
+, colord
+, cups
+, gcr
+, gettext
+, glib
+, glibc
+, gnome-desktop
+, gnome-online-accounts
+, gnome
+, gsound
+, gtk3
+, ibus
+, krb5
+, libepoxy
+, libgnomekbd
+, libgtop
+, libgudev
+, libhandy
+, libnma
+, libpulseaudio
+, libpwquality
+, libsecret
+, libwacom
+, libxml2
+, meson
+, modemmanager
+, networkmanager
+, networkmanagerapplet
+, ninja
+, pkg-config
+, polkit
+, samba
+, shadow
+, tzdata
+, udisks
+, upower
+, wrapGAppsHook
+}:
+
+stdenv.mkDerivation rec {
+  pname = "budgie-control-center";
+  version = "1.2.0";
+
+  src = fetchFromGitHub {
+    owner = "BuddiesOfBudgie";
+    repo = pname;
+    rev = "v${version}";
+    fetchSubmodules = true;
+    sha256 = "sha256-z9apestNLEUKzrCMNo0BNAWeyE6FsUCAzcHIom8LcUs=";
+  };
+
+  patches = [
+    (substituteAll {
+      src = ./paths.patch;
+      budgie_desktop = budgie-desktop;
+      gcm = gnome.gnome-color-manager;
+      inherit cups glibc libgnomekbd shadow;
+      inherit networkmanagerapplet tzdata;
+    })
+  ];
+
+  nativeBuildInputs = [
+    gettext
+    glib
+    meson
+    ninja
+    pkg-config
+    wrapGAppsHook
+  ];
+
+  buildInputs = [
+    accountsservice
+    clutter
+    clutter-gtk
+    colord
+    colord-gtk
+    gcr
+    gnome-desktop
+    gnome-online-accounts
+    gnome.cheese
+    gnome.gnome-bluetooth_1_0
+    gnome.gnome-settings-daemon
+    gsound
+    gtk3
+    ibus
+    krb5
+    libepoxy
+    libgtop
+    libgudev
+    libhandy
+    libnma
+    libpulseaudio
+    libpwquality
+    libsecret
+    libwacom
+    libxml2
+    modemmanager
+    networkmanager
+    polkit
+    samba
+    udisks
+    upower
+  ];
+
+  preConfigure = ''
+    # For ITS rules
+    addToSearchPath "XDG_DATA_DIRS" "${polkit.out}/share"
+  '';
+
+  preFixup = ''
+    gappsWrapperArgs+=(
+      # Sound effects
+      --prefix XDG_DATA_DIRS : "${budgie-desktop}/share"
+    )
```

Comment:
Budgie Control Center crashes when trying to open the keyboard shortcuts editor:

<details>
<summary>Crash log:</summary>
<pre>
machine # [   61.159377] .budgie-control[1385]: Settings schema 'org.gnome.shell.keybindings' is not installed
machine # [   61.418475] show_signal: 153 callbacks suppressed
machine # [   61.418481] traps: .budgie-control[1385] trap int3 ip:7f80d5ca6657 sp:7ffde24a8dc0 error:0 in libglib-2.0.so.0.7400.3[7f80d5c68000+8d000]
machine # [   61.234970] systemd[1]: Created slice Slice /system/systemd-coredump.
machine # [   61.242551] systemd[1]: Started Process Core Dump (PID 1433/UID 0).
machine # [   63.017588] systemd-coredump[1434]: Process 1385 (.budgie-control) of user 1000 dumped core.
machine # 
machine # Module librsvg-2.so.2 without build-id.
machine # Module libpixbufloader-svg.so without build-id.
machine # Module libdconfsettings.so without build-id.
machine # Module libncursesw.so.6 without build-id.
machine # Module libdrm_nouveau.so.2 without build-id.
machine # Module libdrm_amdgpu.so.1 without build-id.
machine # Module libelf.so.0 without build-id.
machine # Module libdrm_radeon.so.1 without build-id.
machine # Module libxcb-xfixes.so.0 without build-id.
machine # Module libxcb-sync.so.1 without build-id.
machine # Module libxcb-present.so.0 without build-id.
machine # Module libxcb-dri3.so.0 without build-id.
machine # Module libxshmfence.so.1 without build-id.
machine # Module libXxf86vm.so.1 without build-id.
machine # Module libxcb-dri2.so.0 without build-id.
machine # Module libxcb-glx.so.0 without build-id.
machine # Module libcluster-samba4.so without build-id.
machine # Module libopus.so.0 without build-id.
machine # Module libvorbisenc.so.2 without build-id.
machine # Module libFLAC.so.12 without build-id.
machine # Module libflag-mapping-samba4.so without build-id.
machine # Module libndr-krb5pac.so.0 without build-id.
machine # Module libcli-ldap-common-samba4.so without build-id.
machine # Module libclidns-samba4.so without build-id.
machine # Module libhcrypto-samba4.so without build-id.
machine # Module libwind-samba4.so without build-id.
machine # Module libheimbase-samba4.so without build-id.
machine # Module libhx509-samba4.so without build-id.
machine # Module libroken-samba4.so without build-id.
machine # Module libmsghdr-samba4.so without build-id.
machine # Module libjansson.so.4 without build-id.
machine # Module libMESSAGING-SEND-samba4.so without build-id.
machine # Module libsamdb.so.0 without build-id.
machine # Module libwbclient.so.0 without build-id.
machine # Module libsamba-modules-samba4.so without build-id.
machine # Module libasn1-samba4.so without build-id.
machine # Module libICE.so.6 without build-id.
machine # Module libSM.so.6 without build-id.
machine # Module libaspell.so.15 without build-id.
machine # Module libsharpyuv.so.0 without build-id.
machine # Module libX11-xcb.so.1 without build-id.
machine # Module libwoff2common.so.1.0.2 without build-id.
machine # Module libogg.so.0 without build-id.
machine # Module libvorbis.so.0 without build-id.
machine # Module liborc-0.4.so.0 without build-id.
machine # Module libevdev.so.2 without build-id.
machine # Module libmtdev.so.1 without build-id.
machine # Module libsndfile.so.1 without build-id.
machine # Module libbrotlicommon.so.1 without build-id.
machine # Module libmd.so.0 without build-id.
machine # Module libunwind-x86_64.so.8 without build-id.
machine # Module libldb.so.2 without build-id.
machine # Module libsamdb-common-samba4.so without build-id.
machine # Module libldbsamba-samba4.so without build-id.
machine # Module libcli-nbt-samba4.so without build-id.
machine # Module libauthkrb5-samba4.so without build-id.
machine # Module libcli-cldap-samba4.so without build-id.
machine # Module libgssapi-samba4.so without build-id.
machine # Module libaddns-samba4.so without build-id.
machine # Module libndr-nbt.so.0 without build-id.
machine # Module libkrb5-samba4.so without build-id.
machine # Module libtdb.so.1 without build-id.
machine # Module libserver-role-samba4.so without build-id.
machine # Module libutil-setid-samba4.so without build-id.
machine # Module libtdb-wrap-samba4.so without build-id.
machine # Module libtime-basic-samba4.so without build-id.
machine # Module libsamba-cluster-support-samba4.so without build-id.
machine # Module libinterfaces-samba4.so without build-id.
machine # Module libmessages-dgm-samba4.so without build-id.
machine # Module libiov-buf-samba4.so without build-id.
machine # Module libtalloc-report-printf-samba4.so without build-id.
machine # Module libserver-id-db-samba4.so without build-id.
machine # Module libsys-rw-samba4.so without build-id.
machine # Module libmessages-util-samba4.so without build-id.
machine # Module libutil-reg-samba4.so without build-id.
machine # Module libsmbd-shim-samba4.so without build-id.
machine # Module libcommon-auth-samba4.so without build-id.
machine # Module libsocket-blocking-samba4.so without build-id.
machine # Module libutil-tdb-samba4.so without build-id.
machine # Module libnpa-tstream-samba4.so without build-id.
machine # Module libsamba-sockets-samba4.so without build-id.
machine # Module libndr-samba-samba4.so without build-id.
machine # Module libdbwrap-samba4.so without build-id.
machine # Module libdcerpc-binding.so.0 without build-id.
machine # Module libsmb-transport-samba4.so without build-id.
machine # Module libcom-err-samba4.so without build-id.
machine # Module libcliauth-samba4.so without build-id.
machine # Module libgensec-samba4.so without build-id.
machine # Module libCHARSET3-samba4.so without build-id.
machine # Module libkrb5samba-samba4.so without build-id.
machine # Module libasn1util-samba4.so without build-id.
machine # Module libssp.so.0 without build-id.
machine # Module libmanette-0.2.so.0 without build-id.
machine # Module libXt.so.6 without build-id.
machine # Module libenchant-2.so.2 without build-id.
machine # Module libwebp.so.7 without build-id.
machine # Module libwebpdemux.so.2 without build-id.
machine # Module libopenjp2.so.7 without build-id.
machine # Module libgstfft-1.0.so.0 without build-id.
machine # Module libgstgl-1.0.so.0 without build-id.
machine # Module libgstapp-1.0.so.0 without build-id.
machine # Module libgstallocators-1.0.so.0 without build-id.
machine # Module libgpg-error.so.0 without build-id.
machine # Module libharfbuzz-icu.so.0 without build-id.
machine # Module libwoff2dec.so.1.0.2 without build-id.
machine # Module libGLESv2.so.2 without build-id.
machine # Module libxslt.so.1 without build-id.
machine # Module libWPEBackend-fdo-1.0.so.1 without build-id.
machine # Module libwpe-1.0.so.1 without build-id.
machine # Module libnghttp2.so.14 without build-id.
machine # Module libpsl.so.5 without build-id.
machine # Module libgcc_s.so.1 without build-id.
machine # Module libstdc++.so.6 without build-id.
machine # Module libicudata.so.72 without build-id.
machine # Module libatomic.so.1 without build-id.
machine # Module liblz4.so.1 without build-id.
machine # Module libzstd.so.1 without build-id.
machine # Module liblzma.so.5 without build-id.
machine # Module libcap.so.2 without build-id.
machine # Module libltdl.so.7 without build-id.
machine # Module libvorbisfile.so.3 without build-id.
machine # Module libgsttag-1.0.so.0 without build-id.
machine # Module libgstbase-1.0.so.0 without build-id.
machine # Module libgstaudio-1.0.so.0 without build-id.
machine # Module libgstvideo-1.0.so.0 without build-id.
machine # Module libunwind.so.8 without build-id.
machine # Module libinput.so.10 without build-id.
machine # Module libdrm.so.2 without build-id.
machine # Module libcogl-pango.so.20 without build-id.
machine # Module libcogl-path.so.20 without build-id.
machine # Module libGLX.so.0 without build-id.
machine # Module libXdmcp.so.6 without build-id.
machine # Module libXau.so.6 without build-id.
machine # Module libGLdispatch.so.0 without build-id.
machine # Module libdatrie.so.1 without build-id.
machine # Module libicui18n.so.72 without build-id.
machine # Module libicuuc.so.72 without build-id.
machine # Module libatspi.so.0 without build-id.
machine # Module libgraphite2.so.3 without build-id.
machine # Module libpcre.so.1 without build-id.
machine # Module libnotify.so.4 without build-id.
machine # Module libgck-2.so.0.0.0 without build-id.
machine # Module libgcr-4.so.0.0.0 without build-id.
machine # Module libkeyutils.so.1 without build-id.
machine # Module libkrb5support.so.0 without build-id.
machine # Module libcom_err.so.3 without build-id.
machine # Module libk5crypto.so.3 without build-id.
machine # Module libcanberra.so.0 without build-id.
machine # Module libdbus-1.so.3 without build-id.
machine # Module libpulsecommon-16.1.so without build-id.
machine # Module libcrack.so.2 without build-id.
machine # Module libgck-1.so.0 without build-id.
machine # Module libgmp.so.10 without build-id.
machine # Module libhogweed.so.6 without build-id.
machine # Module libnettle.so.8 without build-id.
machine # Module libtasn1.so.6 without build-id.
machine # Module libunistring.so.2 without build-id.
machine # Module libidn2.so.0 without build-id.
machine # Module libp11-kit.so.0 without build-id.
machine # Module libgcrypt.so.20 without build-id.
machine # Module libexpat.so.1 without build-id.
machine # Module libbrotlidec.so.1 without build-id.
machine # Module libbz2.so.1 without build-id.
machine # Module libcrypt.so.1 without build-id.
machine # Module libtevent.so.0 without build-id.
machine # Module libtalloc.so.2 without build-id.
machine # Module libbsd.so.0 without build-id.
machine # Module libsecrets3-samba4.so without build-id.
machine # Module libgenrand-samba4.so without build-id.
machine # Module libsamba-debug-samba4.so without build-id.
machine # Module libsamba3-util-samba4.so without build-id.
machine # Module libndr-standard.so.0 without build-id.
machine # Module libsamba-hostconfig.so.0 without build-id.
machine # Module libtevent-util.so.0 without build-id.
machine # Module libsamba-credentials.so.1 without build-id.
machine # Module libdcerpc-samba-samba4.so without build-id.
machine # Module libcli-smb-common-samba4.so without build-id.
machine # Module libgse-samba4.so without build-id.
machine # Module libsamba-errors.so.1 without build-id.
machine # Module libreplace-samba4.so without build-id.
machine # Module libsamba-util.so.0 without build-id.
machine # Module libsmbconf.so.0 without build-id.
machine # Module libmsrpc3-samba4.so without build-id.
machine # Module libsamba-security-samba4.so without build-id.
machine # Module liblibcli-lsa3-samba4.so without build-id.
machine # Module liblibsmb-samba4.so without build-id.
machine # Module libndr.so.3 without build-id.
machine # Module libavahi-client.so.3 without build-id.
machine # Module libavahi-common.so.3 without build-id.
machine # Module libsoup-3.0.so.0 without build-id.
machine # Module libjson-glib-1.0.so.0 without build-id.
machine # Module liblcms2.so.2 without build-id.
machine # Module libseccomp.so.2 without build-id.
machine # Module libudev.so.1 without build-id.
machine # Module libxkbregistry.so.0 without build-id.
machine # Module libsystemd.so.0 without build-id.
machine # Module libcanberra-gtk3.so.0 without build-id.
machine # Module libcogl.so.20 without build-id.
machine # Module libclutter-gtk-1.0.so.0 without build-id.
machine # Module libgstpbutils-1.0.so.0 without build-id.
machine # Module libclutter-gst-3.0.so.0 without build-id.
machine # Module libgstreamer-1.0.so.0 without build-id.
machine # Module libclutter-1.0.so.0 without build-id.
machine # Module libjpeg.so.62 without build-id.
machine # Module libGL.so.1 without build-id.
machine # Module libXrender.so.1 without build-id.
machine # Module libxcb-render.so.0 without build-id.
machine # Module libxcb.so.1 without build-id.
machine # Module libxcb-shm.so.0 without build-id.
machine # Module libpng16.so.16 without build-id.
machine # Module libEGL.so.1 without build-id.
machine # Module libfreetype.so.6 without build-id.
machine # Module libthai.so.0 without build-id.
machine # Module libXinerama.so.1 without build-id.
machine # Module libXrandr.so.2 without build-id.
machine # Module libXcomposite.so.1 without build-id.
machine # Module libXdamage.so.1 without build-id.
machine # Module libXcursor.so.1 without build-id.
machine # Module libXext.so.6 without build-id.
machine # Module libxkbcommon.so.0 without build-id.
machine # Module libXfixes.so.3 without build-id.
machine # Module libtracker-sparql-3.0.so.0 without build-id.
machine # Module libatk-bridge-2.0.so.0 without build-id.
machine # Module libepoxy.so.0 without build-id.
machine # Module libpangoft2-1.0.so.0 without build-id.
machine # Module libharfbuzz.so.0 without build-id.
machine # Module libfribidi.so.0 without build-id.
machine # Module libpcre2-8.so.0 without build-id.
machine # Module libffi.so.8 without build-id.
machine # Module libselinux.so.1 without build-id.
machine # Module libz.so.1 without build-id.
machine # Module libgnome-bluetooth.so.13 without build-id.
machine # Module libmm-glib.so.0 without build-id.
machine # Module libnma.so.0 without build-id.
machine # Module libnm.so.0 without build-id.
machine # Module libkrb5.so.3 without build-id.
machine # Module libgsound.so.0 without build-id.
machine # Module libpulse-mainloop-glib.so.0 without build-id.
machine # Module libpulse.so.0 without build-id.
machine # Module libpwquality.so.1 without build-id.
machine # Module libgcr-base-3.so.1 without build-id.
machine # Module libgnutls.so.30 without build-id.
machine # Module libsecret-1.so.0 without build-id.
machine # Module libfontconfig.so.1 without build-id.
machine # Module libaccountsservice.so.0 without build-id.
machine # Module libsmbclient.so.0 without build-id.
machine # Module libcups.so.2 without build-id.
machine # Module libibus-1.0.so.5 without build-id.
machine # Module libgudev-1.0.so.0 without build-id.
machine # Module libgtop-2.0.so.11 without build-id.
machine # Module libudisks2.so.0 without build-id.
machine # Module libupower-glib.so.3 without build-id.
machine # Module libpolkit-gobject-1.so.0 without build-id.
machine # Module libcolord-gtk.so.1 without build-id.
machine # Module libcolord.so.2 without build-id.
machine # Module libxml2.so.2 without build-id.
machine # Module libwacom.so.9 without build-id.
machine # Module libcheese-gtk.so.25 without build-id.
machine # Module libXi.so.6 without build-id.
machine # Module libcheese.so.8 without build-id.
machine # Module libX11.so.6 without build-id.
machine # Module libatk-1.0.so.0 without build-id.
machine # Module libpango-1.0.so.0 without build-id.
machine # Module libpangocairo-1.0.so.0 without build-id.
machine # Module libhandy-1.so.0 without build-id.
machine # Module .budgie-control-center-wrapped without build-id.
machine # Stack trace of thread 1385:
machine # #0  0x00007f80d5ca6657 g_log_structured_array (libglib-2.0.so.0 + 0x5f657)
machine # #1  0x00007f80d5ca6a7e g_log_default_handler (libglib-2.0.so.0 + 0x5fa7e)
machine # #2  0x00007f80d5ca6ce7 g_logv (libglib-2.0.so.0 + 0x5fce7)
machine # #3  0x00007f80d5ca6f7f g_log (libglib-2.0.so.0 + 0x5ff7f)
machine # #4  0x00007f80d5edebbf g_settings_set_property (libgio-2.0.so.0 + 0xfabbf)
machine # #5  0x00007f80d5d9f6fd object_set_property (libgobject-2.0.so.0 + 0x1c6fd)
machine # #6  0x00007f80d5d9ff88 g_object_new_internal (libgobject-2.0.so.0 + 0x1cf88)
machine # #7  0x00007f80d5da1b53 g_object_new_valist (libgobject-2.0.so.0 + 0x1eb53)
machine # #8  0x00007f80d5da21a9 g_object_new (libgobject-2.0.so.0 + 0x1f1a9)
machine # #9  0x000000000049a093 cc_keyboard_item_load_from_gsettings (.budgie-control-center-wrapped + 0x9a093)
machine # #10 0x000000000049ad6c append_section (.budgie-control-center-wrapped + 0x9ad6c)
machine # #11 0x000000000049b3c5 reload_sections (.budgie-control-center-wrapped + 0x9b3c5)
machine # #12 0x000000000049bafc cc_keyboard_manager_load_shortcuts (.budgie-control-center-wrapped + 0x9bafc)
machine # #13 0x00000000004986c2 cc_keyboard_shortcut_dialog_init (.budgie-control-center-wrapped + 0x986c2)
machine # #14 0x00007f80d5dbc5e9 g_type_create_instance (libgobject-2.0.so.0 + 0x395e9)
machine # #15 0x00007f80d5d9fed0 g_object_new_internal (libgobject-2.0.so.0 + 0x1ced0)
machine # #16 0x00007f80d5da1b53 g_object_new_valist (libgobject-2.0.so.0 + 0x1eb53)
machine # #17 0x00007f80d5da21a9 g_object_new (libgobject-2.0.so.0 + 0x1f1a9)
machine # #18 0x00000000004936c2 keyboard_shortcuts_activated (.budgie-control-center-wrapped + 0x936c2)
machine # #19 0x00007f80d5d9a420 g_closure_invoke (libgobject-2.0.so.0 + 0x17420)
machine # #20 0x00007f80d5dad3d6 signal_emit_unlocked_R.isra.0 (libgobject-2.0.so.0 + 0x2a3d6)
machine # #21 0x00007f80d5db42ed g_signal_emit_valist (libgobject-2.0.so.0 + 0x312ed)
machine # #22 0x00007f80d5db456f g_signal_emit (libgobject-2.0.so.0 + 0x3156f)
machine # #23 0x00007f80d55ab4c9 gtk_list_box_multipress_gesture_released (libgtk-3.so.0 + 0x1f24c9)
machine # #24 0x00007f80d545d3a9 _gtk_marshal_VOID__INT_DOUBLE_DOUBLEv (libgtk-3.so.0 + 0xa43a9)
machine # #25 0x00007f80d5d9a619 _g_closure_invoke_va (libgobject-2.0.so.0 + 0x17619)
machine # #26 0x00007f80d5db42b7 g_signal_emit_valist (libgobject-2.0.so.0 + 0x312b7)
machine # #27 0x00007f80d5db456f g_signal_emit (libgobject-2.0.so.0 + 0x3156f)
machine # #28 0x00007f80d5569006 gtk_gesture_multi_press_end (libgtk-3.so.0 + 0x1b0006)
machine # #29 0x00007f80d5d9d3c2 g_cclosure_marshal_VOID__BOXEDv (libgobject-2.0.so.0 + 0x1a3c2)
machine # #30 0x00007f80d5d9a619 _g_closure_invoke_va (libgobject-2.0.so.0 + 0x17619)
machine # #31 0x00007f80d5db42b7 g_signal_emit_valist (libgobject-2.0.so.0 + 0x312b7)
machine # #32 0x00007f80d5db456f g_signal_emit (libgobject-2.0.so.0 + 0x3156f)
machine # #33 0x00007f80d5566143 _gtk_gesture_check_recognized (libgtk-3.so.0 + 0x1ad143)
machine # #34 0x00007f80d5567723 gtk_gesture_handle_event (libgtk-3.so.0 + 0x1ae723)
machine # #35 0x00007f80d556a7d5 gtk_gesture_single_handle_event (libgtk-3.so.0 + 0x1b17d5)
machine # #36 0x00007f80d5530a81 gtk_event_controller_handle_event (libgtk-3.so.0 + 0x177a81)
machine # #37 0x00007f80d56fa88d _gtk_widget_run_controllers (libgtk-3.so.0 + 0x34188d)
machine # #38 0x00007f80d5456a54 _gtk_marshal_BOOLEAN__BOXEDv (libgtk-3.so.0 + 0x9da54)
machine # #39 0x00007f80d5d9a619 _g_closure_invoke_va (libgobject-2.0.so.0 + 0x17619)
machine # #40 0x00007f80d5db381e g_signal_emit_valist (libgobject-2.0.so.0 + 0x3081e)
machine # #41 0x00007f80d5db456f g_signal_emit (libgobject-2.0.so.0 + 0x3156f)
machine # #42 0x00007f80d56fc214 gtk_widget_event_internal.part.0 (libgtk-3.so.0 + 0x343214)
machine # #43 0x00007f80d55b145e propagate_event (libgtk-3.so.0 + 0x1f845e)
machine # #44 0x00007f80d55b2e26 gtk_main_do_event (libgtk-3.so.0 + 0x1f9e26)
machine # #45 0x00007f80d52ec705 _gdk_event_emit (libgdk-3.so.0 + 0x3f705)
machine # #46 0x00007f80d53449d2 gdk_event_source_dispatch (libgdk-3.so.0 + 0x979d2)
machine # #47 0x00007f80d5c9f609 g_main_context_dispatch (libglib-2.0.so.0 + 0x58609)
machine # #48 0x00007f80d5c9f898 g_main_context_iterate.constprop.0 (libglib-2.0.so.0 + 0x58898)
machine # #49 0x00007f80d5c9f92c g_main_context_iteration (libglib-2.0.so.0 + 0x5892c)
machine # #50 0x00007f80d5ec7e3d g_application_run (libgio-2.0.so.0 + 0xe3e3d)
machine # #51 0x000000000045ccdd main (.budgie-control-center-wrapped + 0x5ccdd)
machine # #52 0x00007f80d3f9d24e __libc_start_call_main (libc.so.6 + 0x2924e)
machine # #53 0x00007f80d3f9d309 __libc_start_main@@GLIBC_2.34 (libc.so.6 + 0x29309)
machine # #54 0x000000000045cf15 _start (.budgie-control-center-wrapped + 0x5cf15)
machine # 
machine # Stack trace of thread 1387:
machine # #0  0x00007f80d4076759 __poll (libc.so.6 + 0x102759)
machine # #1  0x00007f80d5c9f80e g_main_context_iterate.constprop.0 (libglib-2.0.so.0 + 0x5880e)
machine # #2  0x00007f80d5c9f92c g_main_context_iteration (libglib-2.0.so.0 + 0x5892c)
machine # #3  0x00007f80d5c9f971 glib_worker_main (libglib-2.0.so.0 + 0x58971)
machine # #4  0x00007f80d5cc9e4d g_thread_proxy (libglib-2.0.so.0 + 0x82e4d)
machine # #5  0x00007f80d3ffce86 start_thread (libc.so.6 + 0x88e86)
machine # #6  0x00007f80d4083d30 __clone3 (libc.so.6 + 0x10fd30)
machine # 
machine # Stack trace of thread 1388:
machine # #0  0x00007f80d4076759 __poll (libc.so.6 + 0x102759)
machine # #1  0x00007f80d5c9f80e g_main_context_iterate.constprop.0 (libglib-2.0.so.0 + 0x5880e)
machine # #2  0x00007f80d5c9fb4f g_main_loop_run (libglib-2.0.so.0 + 0x58b4f)
machine # #3  0x00007f80d5f02d46 gdbus_shared_thread_func (libgio-2.0.so.0 + 0x11ed46)
machine # #4  0x00007f80d5cc9e4d g_thread_proxy (libglib-2.0.so.0 + 0x82e4d)
machine # #5  0x00007f80d3ffce86 start_thread (libc.so.6 + 0x88e86)
machine # #6  0x00007f80d4083d30 __clone3 (libc.so.6 + 0x10fd30)
machine # 
machine # Stack trace of thread 1389:
machine # #0  0x00007f80d3ff97d5 __futex_abstimed_wait_common (libc.so.6 + 0x857d5)
machine # #1  0x00007f80d3ffc202 pthread_cond_wait@@GLIBC_2.3.2 (libc.so.6 + 0x88202)
machine # #2  0x00007f80c53f1cf9 cnd_wait (swrast_dri.so + 0x10bcf9)
machine # #3  0x00007f80c53a3eab util_queue_thread_func (swrast_dri.so + 0xbdeab)
machine # #4  0x00007f80c53f1c37 impl_thrd_routine (swrast_dri.so + 0x10bc37)
machine # #5  0x00007f80d3ffce86 start_thread (libc.so.6 + 0x88e86)
machine # #6  0x00007f80d4083d30 __clone3 (libc.so.6 + 0x10fd30)
machine # 
machine # Stack trace of thread 1398:
machine # #0  0x00007f80d407bd3d syscall (libc.so.6 + 0x107d3d)
machine # #1  0x00007f80d5cf4690 g_cond_wait_until (libglib-2.0.so.0 + 0xad690)
machine # #2  0x00007f80d5c6d8a1 g_async_queue_pop_intern_unlocked (libglib-2.0.so.0 + 0x268a1)
machine # #3  0x00007f80d5c6de91 g_async_queue_timeout_pop (libglib-2.0.so.0 + 0x26e91)
machine # #4  0x00007f80d5cca6cd g_thread_pool_thread_proxy (libglib-2.0.so.0 + 0x836cd)
machine # #5  0x00007f80d5cc9e4d g_thread_proxy (libglib-2.0.so.0 + 0x82e4d)
machine # #6  0x00007f80d3ffce86 start_thread (libc.so.6 + 0x88e86)
machine # #7  0x00007f80d4083d30 __clone3 (libc.so.6 + 0x10fd30)
machine # 
machine # Stack trace of thread 1399:
machine # #0  0x00007f80d4076759 __poll (libc.so.6 + 0x102759)
machine # #1  0x00007f80d5c9f80e g_main_context_iterate.constprop.0 (libglib-2.0.so.0 + 0x5880e)
machine # #2  0x00007f80d5c9f92c g_main_context_iteration (libglib-2.0.so.0 + 0x5892c)
machine # #3  0x00007f80c404b4ad dconf_gdbus_worker_thread (libdconfsettings.so + 0xb4ad)
machine # #4  0x00007f80d5cc9e4d g_thread_proxy (libglib-2.0.so.0 + 0x82e4d)
machine # #5  0x00007f80d3ffce86 start_thread (libc.so.6 + 0x88e86)
machine # #6  0x00007f80d4083d30 __clone3 (libc.so.6 + 0x10fd30)
machine # ELF object binary architecture: AMD x86-64
</pre>
</details>

Reply:
I tried adding Mutter to the data dir search path (like https://github.com/NixOS/nixpkgs/blob/nixos-unstable/pkgs/desktops/gnome/core/gnome-control-center/default.nix#L160-L161), but it does nothing.

Reply:
> `machine # #9  0x000000000049a093 cc_keyboard_item_load_from_gsettings (.budgie-control-center-wrapped + 0x9a093)`

Probably some settings schema is not installed, try to run `budgie-control-center` in a terminal and see what it says. Adding the corresponding package to buildInputs probably helps.

Reply:
Okay I rebuilt my laptop to Budgie to give this a try. The terminal says:

```
GLib-GIO-ERROR **: 20:42:45.345: Settings schema 'org.gnome.shell.keybindings' is not installed
```

[I tried to get some debug symbols](https://discourse.nixos.org/t/how-to-investigate-gnome-crashing/19726/2). You probably want to apply https://github.com/NixOS/nixpkgs/commit/a78861db45f194b3d829233602dc124c1c257e0d to budgie-control-center as well:

<details>

```gdb
#0  g_log_structured_array (log_level=<optimized out>, fields=0x7ffe30535780, n_fields=4) at ../glib/gmessages.c:563
        writer_func = <optimized out>
        writer_user_data = <optimized out>
        recursion = <optimized out>
        depth = <optimized out>
        __func__ = "g_log_structured_array"
        _g_boolean_var_ = <optimized out>
#1  0x00007fc7cd2dba7e in g_log_default_handler (log_domain=log_domain@entry=0x7fc7cd569b7c "GLib-GIO", log_level=log_level@entry=6, message=message@entry=0x1b3a6c0 "Settings schema 'org.gnome.shell.keybindings' is not installed", unused_data=unused_data@entry=0x0) at ../glib/gmessages.c:3312
        fields = {{key = 0x7fc7cd334fcf "GLIB_OLD_LOG_API", value = 0x7fc7cd33246d, length = -1}, {key = 0x7fc7cd334ef9 "MESSAGE", value = 0x1b3a6c0, length = -1}, {key = 0x7fc7cd334f0c "PRIORITY", value = 0x7fc7cd32e39a, length = -1}, {key = 0x7fc7cd334f66 "GLIB_DOMAIN", value = 0x7fc7cd569b7c, length = -1}}
        n_fields = <optimized out>
#2  0x00007fc7cd2dbce7 in g_logv (log_domain=0x7fc7cd569b7c "GLib-GIO", log_level=G_LOG_LEVEL_ERROR, format=<optimized out>, args=<optimized out>) at ../glib/gmessages.c:1402
        domain = 0x0
        data = 0x0
        depth = <optimized out>
        log_func = 0x7fc7cd2db9c0 <g_log_default_handler>
        domain_fatal_mask = <optimized out>
        masquerade_fatal = 0
        test_level = 6
        was_fatal = <optimized out>
        was_recursion = <optimized out>
        buffer = <optimized out>
        msg = 0x1b3a6c0 "Settings schema 'org.gnome.shell.keybindings' is not installed"
        msg_alloc = 0x1b3a6c0 "Settings schema 'org.gnome.shell.keybindings' is not installed"
        i = 2
        size = <optimized out>
#3  0x00007fc7cd2dbf7f in g_log (log_domain=log_domain@entry=0x7fc7cd569b7c "GLib-GIO", log_level=log_level@entry=G_LOG_LEVEL_ERROR, format=format@entry=0x7fc7cd58d9e0 "Settings schema '%s' is not installed") at ../glib/gmessages.c:1471
        args = {{gp_offset = 32, fp_offset = 48, overflow_arg_area = 0x7ffe305359b0, reg_save_area = 0x7ffe305358f0}}
#4  0x00007fc7cd513bbf in g_settings_set_property (object=0x1b62ea0, prop_id=<optimized out>, value=<optimized out>, pspec=<optimized out>) at ../gio/gsettings.c:596
        default_source = <optimized out>
        schema_id = 0x1b3af60 "org.gnome.shell.keybindings"
        settings = 0x1b62ea0
        __func__ = "g_settings_set_property"
#5  0x00007fc7cd3d46fd in object_set_property (object=object@entry=0x1b62ea0, pspec=0x1532970, value=0x7ffe30535cc0, nqueue=nqueue@entry=0x0, user_specified=<optimized out>) at ../gobject/gobject.c:1794
        inst = <optimized out>
        class = <optimized out>
        pclass = 0xbc5270
        param_id = 2
        __func__ = "object_set_property"
#6  0x00007fc7cd3d4f88 in g_object_new_internal (class=class@entry=0x157c920, params=params@entry=0x7ffe30535bc0, n_params=n_params@entry=1) at ../gobject/gobject.c:2273
        value = <optimized out>
        pspec = <optimized out>
        j = <optimized out>
        user_specified = <optimized out>
        node = 0x11e0030
        nqueue = 0x0
        object = 0x1b62ea0
        i = <optimized out>
        __func__ = "g_object_new_internal"
#7  0x00007fc7cd3d6b53 in g_object_new_valist (object_type=<optimized out>, first_property_name=first_property_name@entry=0x7fc7cd58d821 "schema-id", var_args=var_args@entry=0x7ffe30535e90) at ../gobject/gobject.c:2567
        vtabs_stack = {0xbc2338, 0x0, 0x0, 0x0, 0xbc1958, 0x0, 0x0, 0x0, 0x600000000, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0}
        params = 0x7ffe30535bc0
        values = 0x7ffe30535cc0
        params_stack = {{pspec = 0x1532970, value = 0x7ffe30535cc0}, {pspec = 0x0, value = 0x0}, {pspec = 0x1b223b0, value = 0x7ffe30535ce0}, {pspec = 0x500, value = 0x0}, {pspec = 0x0, value = 0x200000000}, {pspec = 0x0, value = 0x7ffe30535c78}, {pspec = 0xff05fcee, value = 0x8cfcc3c4c7f81e00}, {pspec = 0x1b61260, value = 0x7fc7b41a195e}, {pspec = 0x1a83400, value = 0x1a833f8}, {pspec = 0x7fc7b41a1968, value = 0x950412de}, {pspec = 0x1b5f7d0, value = 0x7fc7cb43ba34 <__gettext_extract_plural+196>}, {pspec = 0x7ffe30535da8, value = 0x7fc7b41a195f}, {pspec = 0x7fc7b41a1969, value = 0x2}, {pspec = 0x0, value = 0x2c7f81e00}, {pspec = 0x7, value = 0x2}, {pspec = 0x0, value = 0x8cfcc3c4c7f81e00}}
        values_stack = {{g_type = 64, data = {{v_int = 28553056, v_uint = 28553056, v_long = 28553056, v_ulong = 28553056, v_int64 = 28553056, v_uint64 = 28553056, v_float = 6.6005866e-38, v_double = 1.410708405338128e-316, v_pointer = 0x1b3af60}, {v_int = 134217728, v_uint = 134217728, v_long = 134217728, v_ulong = 134217728, v_int64 = 134217728, v_uint64 = 134217728, v_float = 3.85185989e-34, v_double = 6.631236846766476e-316, v_pointer = 0x8000000}}}, {g_type = 28550816, data = {{v_int = 24, v_uint = 24, v_long = 24, v_ulong = 24, v_int64 = 24, v_uint64 = 24, v_float = 3.36311631e-44, v_double = 1.1857575500189917e-322, v_pointer = 0x18}, {v_int = 2, v_uint = 2, v_long = 2, v_ulong = 2, v_int64 = 2, v_uint64 = 2, v_float = 2.80259693e-45, v_double = 9.8813129168249309e-324, v_pointer = 0x2}}}, {g_type = 0, data = {{v_int = -884822552, v_uint = 3410144744, v_long = 140496085364200, v_ulong = 140496085364200, v_int64 = 140496085364200, v_uint64 = 140496085364200, v_float = -12758504, v_double = 6.9414289153630379e-310, v_pointer = 0x7fc7cb42ade8 <__gconv_find_transform+360>}, {v_int = 91, v_uint = 91, v_long = 206158430299, v_ulong = 206158430299, v_int64 = 206158430299, v_uint64 = 206158430299, v_float = 1.2751816e-43, v_double = 1.0185579801129304e-312, v_pointer = 0x300000005b}}}, {g_type = 12398120, data = {{v_int = 28550816, v_uint = 28550816, v_long = 28550816, v_ulong = 28550816, v_int64 = 28550816, v_uint64 = 28550816, v_float = 6.59933104e-38, v_double = 1.4105977346334595e-316, v_pointer = 0x1b3a6a0}, {v_int = 0, v_uint = 0, v_long = 0, v_ulong = 0, v_int64 = 0, v_uint64 = 0, v_float = 0, v_double = 0, v_pointer = 0x0}}}, {g_type = 472446402651, data = {{v_int = -940040704, v_uint = 3354926592, v_long = -8287533964332622336, v_ulong = 10159210109376929280, v_int64 = -8287533964332622336, v_uint64 = 10159210109376929280, v_float = -127036, v_double = -4.1139946960261098e-246, v_pointer = 0x8cfcc3c4c7f81e00}, {v_int = 28553648, v_uint = 28553648, v_long = 28553648, v_ulong = 28553648, v_int64 = 28553648, v_uint64 = 28553648, v_float = 6.60091843e-38, v_double = 1.4107376540243618e-316, v_pointer = 0x1b3b1b0}}}, {g_type = 0, data = {{v_int = 810769904, v_uint = 810769904, v_long = 140729709190640, v_ulong = 140729709190640, v_int64 = 140729709190640, v_uint64 = 140729709190640, v_float = 7.6894846e-10, v_double = 6.9529714660324361e-310, v_pointer = 0x7ffe30535df0}, {v_int = 28550816, v_uint = 28550816, v_long = 28550816, v_ulong = 28550816, v_int64 = 28550816, v_uint64 = 28550816, v_float = 6.59933104e-38, v_double = 1.4105977346334595e-316, v_pointer = 0x1b3a6a0}}}, {g_type = 28550784, data = {{v_int = 1, v_uint = 1, v_long = 1, v_ulong = 1, v_int64 = 1, v_uint64 = 1, v_float = 1.40129846e-45, v_double = 4.9406564584124654e-324, v_pointer = 0x1}, {v_int = -883195529, v_uint = 3411771767, v_long = 140496086991223, v_ulong = 140496086991223, v_int64 = 140496086991223, v_uint64 = 140496086991223, v_float = -14385527, v_double = 6.9414289957486548e-310, v_pointer = 0x7fc7cb5b8177}}}, {g_type = 140496085359041, data = {{v_int = -1, v_uint = 4294967295, v_long = -1, v_ulong = 18446744073709551615, v_int64 = -1, v_uint64 = 18446744073709551615, v_float = -nan(0x7fffff), v_double = -nan(0xfffffffffffff), v_pointer = 0xffffffffffffffff}, {v_int = -940040704, v_uint = 3354926592, v_long = -8287533964332622336, v_ulong = 10159210109376929280, v_int64 = -8287533964332622336, v_uint64 = 10159210109376929280, v_float = -127036, v_double = -4.1139946960261098e-246, v_pointer = 0x8cfcc3c4c7f81e00}}}, {g_type = 0, data = {{v_int = -940040704, v_uint = 3354926592, v_long = -8287533964332622336, v_ulong = 10159210109376929280, v_int64 = -8287533964332622336, v_uint64 = 10159210109376929280, v_float = -127036, v_double = -4.1139946960261098e-246, v_pointer = 0x8cfcc3c4c7f81e00}, {v_int = 2, v_uint = 2, v_long = 2, v_ulong = 2, v_int64 = 2, v_uint64 = 2, v_float = 2.80259693e-45, v_double = 9.8813129168249309e-324, v_pointer = 0x2}}}, {g_type = 18446744073709551416, data = {{v_int = 2, v_uint = 2, v_long = 2, v_ulong = 2, v_int64 = 2, v_uint64 = 2, v_float = 2.80259693e-45, v_double = 9.8813129168249309e-324, v_pointer = 0x2}, {v_int = -1, v_uint = 4294967295, v_long = -1, v_ulong = 18446744073709551615, v_int64 = -1, v_uint64 = 18446744073709551615, v_float = -nan(0x7fffff), v_double = -nan(0xfffffffffffff), v_pointer = 0xffffffffffffffff}}}, {g_type = 28553648, data = {{v_int = 27800384, v_uint = 27800384, v_long = 27800384, v_ulong = 27800384, v_int64 = 27800384, v_uint64 = 27800384, v_float = 6.17869936e-38, v_double = 1.3735214675594657e-316, v_pointer = 0x1a83340}, {v_int = 220, v_uint = 220, v_long = 220, v_ulong = 220, v_int64 = 220, v_uint64 = 220, v_float = 3.08285662e-43, v_double = 1.0869444208507424e-321, v_pointer = 0xdc}}}, {g_type = 140496085815971, data = {{v_int = -1, v_uint = 4294967295, v_long = -1, v_ulong = 18446744073709551615, v_int64 = -1, v_uint64 = 18446744073709551615, v_float = -nan(0x7fffff), v_double = -nan(0xfffffffffffff), v_pointer = 0xffffffffffffffff}, {v_int = 810770048, v_uint = 810770048, v_long = 140729709190784, v_ulong = 140729709190784, v_int64 = 140729709190784, v_uint64 = 140729709190784, v_float = 7.68956454e-10, v_double = 6.9529714660395507e-310, v_pointer = 0x7ffe30535e80}}}, {g_type = 140729709190864, data = {{v_int = -1273351727, v_uint = 3021615569, v_long = 140495696835025, v_ulong = 140495696835025, v_int64 = 140495696835025, v_uint64 = 140495696835025, v_float = -1.43604908e-07, v_double = 6.9414097194712605e-310, v_pointer = 0x7fc7b41a31d1}, {v_int = 810770128, v_uint = 810770128, v_long = 140729709190864, v_ulong = 140729709190864, v_int64 = 140729709190864, v_uint64 = 140729709190864, v_float = 7.68960895e-10, v_double = 6.9529714660435032e-310, v_pointer = 0x7ffe30535ed0}}}, {g_type = 140496085812007, data = {{v_int = 759583829, v_uint = 759583829, v_long = 241277752405, v_ulong = 241277752405, v_int64 = 241277752405, v_uint64 = 241277752405, v_float = 1.12737224e-11, v_double = 1.192070485691007e-312, v_pointer = 0x382d465455}, {v_int = 80, v_uint = 80, v_long = 80, v_ulong = 80, v_int64 = 80, v_uint64 = 80, v_float = 1.12103877e-43, v_double = 3.9525251667299724e-322, v_pointer = 0x50}}}, {g_type = 1, data = {{v_int = 810770344, v_uint = 810770344, v_long = 13695672232, v_ulong = 13695672232, v_int64 = 13695672232, v_uint64 = 13695672232, v_float = 7.68972885e-10, v_double = 6.7665611465331066e-314, v_pointer = 0x330535fa8}, {v_int = 7, v_uint = 7, v_long = 7, v_ulong = 7, v_int64 = 7, v_uint64 = 7, v_float = 9.80908925e-45, v_double = 3.4584595208887258e-323, v_pointer = 0x7}}}, {g_type = 48, data = {{v_int = 32, v_uint = 32, v_long = 32, v_ulong = 32, v_int64 = 32, v_uint64 = 32, v_float = 4.48415509e-44, v_double = 1.5810100666919889e-322, v_pointer = 0x20}, {v_int = -884374745, v_uint = 3410592551, v_long = 140496085812007, v_ulong = 140496085812007, v_int64 = 140496085812007, v_uint64 = 140496085812007, v_float = -13206311, v_double = 6.9414289374876434e-310, v_pointer = 0x7fc7cb498327 <_int_malloc+3223>}}}}
        name = <optimized out>
        vtabs = 0x7ffe30535b40
        n_params = <optimized out>
        n_params_alloc = 16
        class = 0x157c920
        unref_class = <optimized out>
        object = <optimized out>
        __func__ = "g_object_new_valist"
#8  0x00007fc7cd3d71a9 in g_object_new (object_type=<optimized out>, first_property_name=first_property_name@entry=0x7fc7cd58d821 "schema-id") at ../gobject/gobject.c:2040
        object = <optimized out>
        var_args = {{gp_offset = 32, fp_offset = 48, overflow_arg_area = 0x7ffe30535f70, reg_save_area = 0x7ffe30535eb0}}
#9  0x00007fc7cd514cf5 in g_settings_new (schema_id=<optimized out>) at ../gio/gsettings.c:984
        __func__ = "g_settings_new"
#10 0x000000000049a093 in cc_keyboard_item_load_from_gsettings (item=item@entry=0x1b60cb0, description=0x1b625c0 "Activate favorite application 1", schema=<optimized out>, key=0x1b63e20 "switch-to-application-1") at ../panels/keyboard/cc-keyboard-item.c:567
        signal_name = 0x0
        variant = 0x0
#11 0x000000000049ad6c in append_section (self=0x18106d0, title=0x7fc7b41a31d1 "启动器", id=0x1b5ebf0 "Launchers", group=BINDING_GROUP_SYSTEM, keys_list=0x1b3b9d0) at ../panels/keyboard/cc-keyboard-manager.c:315
        item = 0x1b60cb0
        ret = <optimized out>
        iter = {stamp = 0, user_data = 0xffffffffffffff38, user_data2 = 0x2, user_data3 = 0x1b3af60}
        reverse_items = 0x177b1e0
        hash = 0x177d5e0
        keys_array = 0xf245a0
        is_new = <optimized out>
        i = 0
        __func__ = "append_section"
#12 0x000000000049b3c5 in append_sections_from_file (wm_keybindings=0x1b15ce0, datadir=<optimized out>, path=0x1b3a360 "/nix/store/qpaiq2ka4mmaps77ffvif8x54cykgwa4-gnome-shell-43.2/share/gnome-control-center/keybindings/50-gnome-shell-launchers.xml", self=0x18106d0) at ../panels/keyboard/cc-keyboard-manager.c:431
        keylist = 0x1b61680
        keys = 0x1b3b9d0
        key = {type = CC_KEYBOARD_ITEM_TYPE_NONE, schema = 0x0, description = 0x0, name = 0x0, reverse_entry = 0x0, is_reversed = 0, hidden = 0}
        group = <optimized out>
        i = <optimized out>
        title = <optimized out>
        keylist = <optimized out>
        keys = <optimized out>
        key = <optimized out>
        title = <optimized out>
        group = <optimized out>
        i = <optimized out>
        localedir = <optimized out>
        entry = <optimized out>
#13 reload_sections (self=self@entry=0x18106d0) at ../panels/keyboard/cc-keyboard-manager.c:581
        path = 0x1b3a360 "/nix/store/qpaiq2ka4mmaps77ffvif8x54cykgwa4-gnome-shell-43.2/share/gnome-control-center/keybindings/50-gnome-shell-launchers.xml"
        dir_path = 0x1b3a2d0 "/nix/store/qpaiq2ka4mmaps77ffvif8x54cykgwa4-gnome-shell-43.2/share/gnome-control-center/keybindings"
        name = <optimized out>
        loaded_files = 0x177d700
        dir = 0x1b61e80
        default_wm_keybindings = {0x58703c "Mutter", 0x0}
        wm_keybindings = 0x1b15ce0
        data_dirs = <optimized out>
        i = <optimized out>
        pass = <optimized out>
#14 0x000000000049bafc in cc_keyboard_manager_load_shortcuts (self=0x18106d0) at ../panels/keyboard/cc-keyboard-manager.c:719
        __func__ = "cc_keyboard_manager_load_shortcuts"
#15 0x00000000004986c2 in cc_keyboard_shortcut_dialog_init (self=0x15f66d0) at ../panels/keyboard/cc-keyboard-shortcut-dialog.c:844
No locals.
#16 0x00007fc7cd3f15e9 in g_type_create_instance (type=<optimized out>) at ../gobject/gtype.c:1931
        node = 0xeba4f0
        instance = 0x15f66d0
        class = 0x1ae8be0
        allocated = <optimized out>
        private_size = <optimized out>
        ivar_size = <optimized out>
        i = <optimized out>
#17 0x00007fc7cd3d4ed0 in g_object_new_internal (class=class@entry=0x1ae8be0, params=params@entry=0x7ffe30536460, n_params=n_params@entry=1) at ../gobject/gobject.c:2228
        nqueue = 0x0
        object = <optimized out>
        i = <optimized out>
        __func__ = "g_object_new_internal"
#18 0x00007fc7cd3d6b53 in g_object_new_valist (object_type=<optimized out>, first_property_name=first_property_name@entry=0x5867f4 "use-header-bar", var_args=var_args@entry=0x7ffe30536730) at ../gobject/gobject.c:2567
        vtabs_stack = {0xbc1958, 0x4c, 0x7ffe30536390, 0x1, 0x10, 0x10, 0x1, 0x0, 0x7ffe305364c0, 0xae1, 0x19441c0, 0xc7a340, 0x1, 0x4, 0x0, 0x1928b20}
        params = 0x7ffe30536460
        values = 0x7ffe30536560
        params_stack = {{pspec = 0x1562c50, value = 0x7ffe30536560}, {pspec = 0xc76d40, value = 0x0}, {pspec = 0x0, value = 0x0}, {pspec = 0x0, value = 0x0}, {pspec = 0x0, value = 0x0}, {pspec = 0x14f8e10, value = 0x8cfcc3c4c7f81e00}, {pspec = 0x19441b0, value = 0x7ffe30536618}, {pspec = 0xc76d40, value = 0x7ffe30536618}, {pspec = 0x7ffe30536610, value = 0x0}, {pspec = 0x19441c0, value = 0x7fc7cd3e956f <g_signal_emit+143>}, {pspec = 0x3000000020, value = 0x7ffe305365e0}, {pspec = 0x7ffe30536520, value = 0x8cfcc3c4c7f81e00}, {pspec = 0x1688240, value = 0x7ffe30536570}, {pspec = 0xbbc700, value = 0xc76d40}, {pspec = 0x0, value = 0x10}, {pspec = 0x8, value = 0x10}}
        values_stack = {{g_type = 24, data = {{v_int = 1, v_uint = 1, v_long = 1, v_ulong = 1, v_int64 = 1, v_uint64 = 1, v_float = 1.40129846e-45, v_double = 4.9406564584124654e-324, v_pointer = 0x1}, {v_int = 0, v_uint = 0, v_long = 0, v_ulong = 0, v_int64 = 0, v_uint64 = 0, v_float = 0, v_double = 0, v_pointer = 0x0}}}, {g_type = 31, data = {{v_int = -851745664, v_uint = 3443221632, v_long = 140496118441088, v_ulong = 140496118441088, v_int64 = 140496118441088, v_uint64 = 140496118441088, v_float = -196495360, v_double = 6.9414305495784411e-310, v_pointer = 0x7fc7cd3b6480 <private_thread_memory>}, {v_int = -851745664, v_uint = 3443221632, v_long = 140496118441088, v_ulong = 140496118441088, v_int64 = 140496118441088, v_uint64 = 140496118441088, v_float = -196495360, v_double = 6.9414305495784411e-310, v_pointer = 0x7fc7cd3b6480 <private_thread_memory>}}}, {g_type = 56, data = {{v_int = -852545856, v_uint = 3442421440, v_long = 140496117640896, v_ulong = 140496117640896, v_int64 = 140496117640896, v_uint64 = 140496117640896, v_float = -183692288, v_double = 6.9414305100437034e-310, v_pointer = 0x7fc7cd2f2ec0 <g_slice_alloc+32>}, {v_int = 26493392, v_uint = 26493392, v_long = 26493392, v_ulong = 26493392, v_int64 = 26493392, v_uint64 = 26493392, v_float = 5.446105e-38, v_double = 1.3089474829005314e-316, v_pointer = 0x19441d0}}}, {g_type = 26625440, data = {{v_int = 0, v_uint = 0, v_long = 0, v_ulong = 0, v_int64 = 0, v_uint64 = 0, v_float = 0, v_double = 0, v_pointer = 0x0}, {v_int = 810772304, v_uint = 810772304, v_long = 140729709193040, v_ulong = 140729709193040, v_int64 = 140729709193040, v_uint64 = 140729709193040, v_float = 7.69081687e-10, v_double = 6.9529714661510119e-310, v_pointer = 0x7ffe30536750}}}, {g_type = 140729709193024, data = {{v_int = -884374745, v_uint = 3410592551, v_long = 140496085812007, v_ulong = 140496085812007, v_int64 = 140496085812007, v_uint64 = 140496085812007, v_float = -13206311, v_double = 6.9414289374876434e-310, v_pointer = 0x7fc7cb498327 <_int_malloc+3223>}, {v_int = 56, v_uint = 56, v_long = 56, v_ulong = 56, v_int64 = 56, v_uint64 = 56, v_float = 7.8472714e-44, v_double = 2.7667676167109806e-322, v_pointer = 0x38}}}, {g_type = 160, data = {{v_int = 2, v_uint = 2, v_long = 2, v_ulong = 2, v_int64 = 2, v_uint64 = 2, v_float = 2.80259693e-45, v_double = 9.8813129168249309e-324, v_pointer = 0x2}, {v_int = 22438664, v_uint = 22438664, v_long = 25792242440, v_ulong = 25792242440, v_int64 = 25792242440, v_uint64 = 25792242440, v_float = 3.93766438e-38, v_double = 1.2743060918812609e-313, v_pointer = 0x601566308}}}, {g_type = 7, data = {{v_int = 96, v_uint = 96, v_long = 96, v_ulong = 96, v_int64 = 96, v_uint64 = 96, v_float = 1.34524653e-43, v_double = 4.7430302000759668e-322, v_pointer = 0x60}, {v_int = 80, v_uint = 80, v_long = 80, v_ulong = 80, v_int64 = 80, v_uint64 = 80, v_float = 1.12103877e-43, v_double = 3.9525251667299724e-322, v_pointer = 0x50}}}, {g_type = 5156723102014480515, data = {{v_int = 12116016, v_uint = 12116016, v_long = 12116016, v_ulong = 12116016, v_int64 = 12116016, v_uint64 = 12116016, v_float = 1.69781546e-38, v_double = 5.9861072700628766e-317, v_pointer = 0xb8e030}, {v_int = 128, v_uint = 128, v_long = 128, v_ulong = 128, v_int64 = 128, v_uint64 = 128, v_float = 1.79366203e-43, v_double = 6.3240402667679558e-322, v_pointer = 0x80}}}, {g_type = 18446744073709551472, data = {{v_int = 1, v_uint = 1, v_long = 1, v_ulong = 1, v_int64 = 1, v_uint64 = 1, v_float = 1.40129846e-45, v_double = 4.9406564584124654e-324, v_pointer = 0x1}, {v_int = 7540584, v_uint = 7540584, v_long = 210460938088, v_ulong = 210460938088, v_int64 = 210460938088, v_uint64 = 210460938088, v_float = 1.05666088e-38, v_double = 1.0398151930080232e-312, v_pointer = 0x3100730f68}}}, {g_type = 0, data = {{v_int = 0, v_uint = 0, v_long = 0, v_ulong = 0, v_int64 = 0, v_uint64 = 0, v_float = 0, v_double = 0, v_pointer = 0x0}, {v_int = 0, v_uint = 0, v_long = 0, v_ulong = 0, v_int64 = 0, v_uint64 = 0, v_float = 0, v_double = 0, v_pointer = 0x0}}}, {g_type = 472446402651, data = {{v_int = 119, v_uint = 119, v_long = 532575944823, v_ulong = 532575944823, v_int64 = 532575944823, v_uint64 = 532575944823, v_float = 1.66754517e-43, v_double = 2.6312747813848758e-312, v_pointer = 0x7c00000077}, {v_int = 0, v_uint = 0, v_long = 0, v_ulong = 0, v_int64 = 0, v_uint64 = 0, v_float = 0, v_double = 0, v_pointer = 0x0}}}, {g_type = 80, data = {{v_int = 80, v_uint = 80, v_long = 80, v_ulong = 80, v_int64 = 80, v_uint64 = 80, v_float = 1.12103877e-43, v_double = 3.9525251667299724e-322, v_pointer = 0x50}, {v_int = -882922848, v_uint = 3412044448, v_long = 140496087263904, v_ulong = 140496087263904, v_int64 = 140496087263904, v_uint64 = 140496087263904, v_float = -14658208, v_double = 6.9414290092208863e-310, v_pointer = 0x7fc7cb5faaa0 <main_arena>}}}, {g_type = 120912, data = {{v_int = 28272560, v_uint = 28272560, v_long = 28272560, v_ulong = 28272560, v_int64 = 28272560, v_uint64 = 28272560, v_float = 6.44336316e-38, v_double = 1.3968500615985393e-316, v_pointer = 0x1af67b0}, {v_int = 0, v_uint = 0, v_long = 0, v_ulong = 0, v_int64 = 0, v_uint64 = 0, v_float = 0, v_double = 0, v_pointer = 0x0}}}, {g_type = 140496085818690, data = {{v_int = 80, v_uint = 80, v_long = 80, v_ulong = 80, v_int64 = 80, v_uint64 = 80, v_float = 1.12103877e-43, v_double = 3.9525251667299724e-322, v_pointer = 0x50}, {v_int = 810772336, v_uint = 810772336, v_long = 140729709193072, v_ulong = 140729709193072, v_int64 = 140729709193072, v_uint64 = 140729709193072, v_float = 7.69083464e-10, v_double = 6.9529714661525929e-310, v_pointer = 0x7ffe30536770}}}, {g_type = 0, data = {{v_int = 1, v_uint = 1, v_long = 1, v_ulong = 1, v_int64 = 1, v_uint64 = 1, v_float = 1.40129846e-45, v_double = 4.9406564584124654e-324, v_pointer = 0x1}, {v_int = 12332048, v_uint = 12332048, v_long = 12332048, v_ulong = 12332048, v_int64 = 12332048, v_uint64 = 12332048, v_float = 1.72808799e-38, v_double = 6.0928412596652528e-317, v_pointer = 0xbc2c10}}}, {g_type = 140496117540161, data = {{v_int = 15443184, v_uint = 15443184, v_long = 15443184, v_ulong = 15443184, v_int64 = 15443184, v_uint64 = 15443184, v_float = 2.164051e-38, v_double = 7.6299466768052052e-317, v_pointer = 0xeba4f0}, {v_int = -851520665, v_uint = 3443446631, v_long = 140496118666087, v_ulong = 140496118666087, v_int64 = 140496118666087, v_uint64 = 140496118666087, v_float = -200095344, v_double = 6.9414305606948688e-310, v_pointer = 0x7fc7cd3ed367 <type_data_make_W+199>}}}}
        name = <optimized out>
        vtabs = 0x7ffe305363e0
        n_params = <optimized out>
        n_params_alloc = 16
        class = 0x1ae8be0
        unref_class = <optimized out>
        object = <optimized out>
        __func__ = "g_object_new_valist"
#19 0x00007fc7cd3d71a9 in g_object_new (object_type=<optimized out>, first_property_name=first_property_name@entry=0x5867f4 "use-header-bar") at ../gobject/gobject.c:2040
        object = <optimized out>
        var_args = {{gp_offset = 32, fp_offset = 48, overflow_arg_area = 0x7ffe30536810, reg_save_area = 0x7ffe30536750}}
#20 0x0000000000499145 in cc_keyboard_shortcut_dialog_new () at ../panels/keyboard/cc-keyboard-shortcut-dialog.c:871
No locals.
#21 0x00000000004936c2 in keyboard_shortcuts_activated (button=<optimized out>, row=<optimized out>, self=0x13d1830) at ../panels/keyboard/cc-keyboard-panel.c:149
        window = 0xd9a340
        shortcut_dialog = <optimized out>
        window = <optimized out>
        shortcut_dialog = <optimized out>
#22 keyboard_shortcuts_activated (button=<optimized out>, row=<optimized out>, self=0x13d1830) at ../panels/keyboard/cc-keyboard-panel.c:138
        window = <optimized out>
        shortcut_dialog = <optimized out>
#23 0x00007fc7cd3cf420 in g_closure_invoke (closure=0x1142d40, return_value=return_value@entry=0x0, n_param_values=2, param_values=param_values@entry=0x7ffe305369c0, invocation_hint=invocation_hint@entry=0x7ffe30536940) at ../gobject/gclosure.c:832
        marshal = 0x7fc7cd3d2570 <g_cclosure_marshal_VOID__OBJECT>
        marshal_data = 0x0
        in_marshal = 0
        real_closure = 0x1142d20
        __func__ = "g_closure_invoke"
#24 0x00007fc7cd3e23d6 in signal_emit_unlocked_R (node=node@entry=0x1503cc0, detail=detail@entry=0, instance=instance@entry=0x150b270, emission_return=emission_return@entry=0x0, instance_and_params=instance_and_params@entry=0x7ffe305369c0) at ../gobject/gsignal.c:3796
        tmp = <optimized out>
        handler = 0x7fc7b00502c0
        accumulator = 0x0
        emission = {next = 0x7ffe30536da0, instance = 0x150b270, ihint = {signal_id = 310, detail = 0, run_type = (G_SIGNAL_RUN_FIRST | G_SIGNAL_ACCUMULATOR_FIRST_RUN)}, state = EMISSION_RUN, chain_type = 4}
        class_closure = 0x1503c70
        hlist = <optimized out>
        handler_list = 0x7fc7b004ec00
        return_accu = 0x0
        accu = {g_type = 0, data = {{v_int = 0, v_uint = 0, v_long = 0, v_ulong = 0, v_int64 = 0, v_uint64 = 0, v_float = 0, v_double = 0, v_pointer = 0x0}, {v_int = 0, v_uint = 0, v_long = 0, v_ulong = 0, v_int64 = 0, v_uint64 = 0, v_float = 0, v_double = 0, v_pointer = 0x0}}}
        signal_id = 310
        max_sequential_handler_number = 2881
        return_value_altered = <optimized out>
        EMIT_RESTART = <optimized out>
#25 0x00007fc7cd3e92ed in g_signal_emit_valist (instance=<optimized out>, signal_id=<optimized out>, detail=<optimized out>, var_args=var_args@entry=0x7ffe30536b60) at ../gobject/gsignal.c:3549
        instance_and_params = 0x7ffe305369c0
        signal_return_type = <optimized out>
        param_values = 0x7ffe305369d8
        node = <optimized out>
        i = <optimized out>
        n_params = <optimized out>
        __func__ = "g_signal_emit_valist"
#26 0x00007fc7cd3e956f in g_signal_emit (instance=<optimized out>, signal_id=<optimized out>, detail=<optimized out>) at ../gobject/gsignal.c:3606
        var_args = {{gp_offset = 32, fp_offset = 48, overflow_arg_area = 0x7ffe30536c40, reg_save_area = 0x7ffe30536b80}}
#27 0x00007fc7cc9f24c9 in gtk_list_box_multipress_gesture_released () from /nix/store/2dm9a1d865d4s49w8p1lxfwwp7gpz2mp-gtk+3-3.24.36/lib/libgtk-3.so.0
No symbol table info available.
#28 0x00007fc7cc8a43a9 in _gtk_marshal_VOID__INT_DOUBLE_DOUBLEv () from /nix/store/2dm9a1d865d4s49w8p1lxfwwp7gpz2mp-gtk+3-3.24.36/lib/libgtk-3.so.0
No symbol table info available.
#29 0x00007fc7cd3cf619 in _g_closure_invoke_va (closure=closure@entry=0x16ab960, return_value=return_value@entry=0x0, instance=instance@entry=0x16a9450, args=args@entry=0x7ffe30536ea0, n_params=3, param_types=0x113bd80) at ../gobject/gclosure.c:895
        marshal = 0x7fc7cc8a4310 <_gtk_marshal_VOID__INT_DOUBLE_DOUBLEv>
        marshal_data = 0x0
        in_marshal = 0
        real_closure = 0x16ab940
        __func__ = "_g_closure_invoke_va"
#30 0x00007fc7cd3e92b7 in g_signal_emit_valist (instance=0x16a9450, signal_id=253, detail=<optimized out>, var_args=var_args@entry=0x7ffe30536ea0) at ../gobject/gsignal.c:3456
        return_accu = <optimized out>
        accu = {g_type = 0, data = {{v_int = 0, v_uint = 0, v_long = 0, v_ulong = 0, v_int64 = 0, v_uint64 = 0, v_float = 0, v_double = 0, v_pointer = 0x0}, {v_int = 0, v_uint = 0, v_long = 0, v_ulong = 0, v_int64 = 0, v_uint64 = 0, v_float = 0, v_double = 0, v_pointer = 0x0}}}
        accumulator = 0x0
        emission = {next = 0x7ffe30537120, instance = 0x16a9450, ihint = {signal_id = 253, detail = 0, run_type = (G_SIGNAL_RUN_FIRST | G_SIGNAL_ACCUMULATOR_FIRST_RUN)}, state = EMISSION_RUN, chain_type = 18569696}
        instance_type = <optimized out>
        emission_return = {g_type = 0, data = {{v_int = 0, v_uint = 0, v_long = 0, v_ulong = 0, v_int64 = 0, v_uint64 = 0, v_float = 0, v_double = 0, v_pointer = 0x0}, {v_int = 0, v_uint = 0, v_long = 0, v_ulong = 0, v_int64 = 0, v_uint64 = 0, v_float = 0, v_double = 0, v_pointer = 0x0}}}
        rtype = 4
        static_scope = 0
        fastpath_handler = <optimized out>
        closure = <optimized out>
        run_type = <optimized out>
        hlist = <optimized out>
        l = <optimized out>
        fastpath = <optimized out>
        instance_and_params = <optimized out>
        signal_return_type = <optimized out>
        param_values = <optimized out>
        node = <optimized out>
        i = <optimized out>
        n_params = <optimized out>
        __func__ = "g_signal_emit_valist"
#31 0x00007fc7cd3e956f in g_signal_emit (instance=<optimized out>, signal_id=<optimized out>, detail=<optimized out>) at ../gobject/gsignal.c:3606
        var_args = {{gp_offset = 24, fp_offset = 48, overflow_arg_area = 0x7ffe30536f80, reg_save_area = 0x7ffe30536ec0}}
#32 0x00007fc7cc9b0006 in gtk_gesture_multi_press_end () from /nix/store/2dm9a1d865d4s49w8p1lxfwwp7gpz2mp-gtk+3-3.24.36/lib/libgtk-3.so.0
No symbol table info available.
#33 0x00007fc7cd3d23c2 in g_cclosure_marshal_VOID__BOXEDv (closure=0x1021ae0, return_value=<optimized out>, instance=0x16a9450, args=<optimized out>, marshal_data=<optimized out>, n_params=<optimized out>, param_types=0x1021b10) at ../gobject/gmarshal.c:1686
        cc = <optimized out>
        data1 = <optimized out>
        data2 = <optimized out>
        callback = 0x7fc7cc9aff60 <gtk_gesture_multi_press_end>
        arg0 = 0x0
        args_copy = {{gp_offset = 32, fp_offset = 48, overflow_arg_area = 0x7ffe30537300, reg_save_area = 0x7ffe30537240}}
#34 0x00007fc7cd3cf619 in _g_closure_invoke_va (closure=closure@entry=0x1021ae0, return_value=return_value@entry=0x0, instance=instance@entry=0x16a9450, args=args@entry=0x7ffe30537220, n_params=1, param_types=0x1021b10) at ../gobject/gclosure.c:895
        marshal = 0x7fc7cd3cdbc0 <g_type_class_meta_marshalv>
        marshal_data = 0x108
        in_marshal = 0
        real_closure = 0x1021ac0
        __func__ = "_g_closure_invoke_va"
#35 0x00007fc7cd3e92b7 in g_signal_emit_valist (instance=0x16a9450, signal_id=239, detail=<optimized out>, var_args=var_args@entry=0x7ffe30537220) at ../gobject/gsignal.c:3456
        return_accu = <optimized out>
        accu = {g_type = 0, data = {{v_int = 0, v_uint = 0, v_long = 0, v_ulong = 0, v_int64 = 0, v_uint64 = 0, v_float = 0, v_double = 0, v_pointer = 0x0}, {v_int = 0, v_uint = 0, v_long = 0, v_ulong = 0, v_int64 = 0, v_uint64 = 0, v_float = 0, v_double = 0, v_pointer = 0x0}}}
        accumulator = 0x0
        emission = {next = 0x7ffe30537570, instance = 0x16a9450, ihint = {signal_id = 239, detail = 0, run_type = (G_SIGNAL_RUN_LAST | G_SIGNAL_ACCUMULATOR_FIRST_RUN)}, state = EMISSION_RUN, chain_type = 18569696}
        instance_type = <optimized out>
        emission_return = {g_type = 0, data = {{v_int = 0, v_uint = 0, v_long = 0, v_ulong = 0, v_int64 = 0, v_uint64 = 0, v_float = 0, v_double = 0, v_pointer = 0x0}, {v_int = 0, v_uint = 0, v_long = 0, v_ulong = 0, v_int64 = 0, v_uint64 = 0, v_float = 0, v_double = 0, v_pointer = 0x0}}}
        rtype = 4
        static_scope = 0
        fastpath_handler = <optimized out>
        closure = <optimized out>
        run_type = <optimized out>
        hlist = <optimized out>
        l = <optimized out>
        fastpath = <optimized out>
        instance_and_params = <optimized out>
        signal_return_type = <optimized out>
        param_values = <optimized out>
        node = <optimized out>
        i = <optimized out>
        n_params = <optimized out>
        __func__ = "g_signal_emit_valist"
#36 0x00007fc7cd3e956f in g_signal_emit (instance=<optimized out>, signal_id=<optimized out>, detail=<optimized out>) at ../gobject/gsignal.c:3606
        var_args = {{gp_offset = 24, fp_offset = 48, overflow_arg_area = 0x7ffe30537300, reg_save_area = 0x7ffe30537240}}
#37 0x00007fc7cc9ad143 in _gtk_gesture_check_recognized () from /nix/store/2dm9a1d865d4s49w8p1lxfwwp7gpz2mp-gtk+3-3.24.36/lib/libgtk-3.so.0
No symbol table info available.
#38 0x00007fc7cc9ae723 in gtk_gesture_handle_event () from /nix/store/2dm9a1d865d4s49w8p1lxfwwp7gpz2mp-gtk+3-3.24.36/lib/libgtk-3.so.0
No symbol table info available.
#39 0x00007fc7cc9b17d5 in gtk_gesture_single_handle_event () from /nix/store/2dm9a1d865d4s49w8p1lxfwwp7gpz2mp-gtk+3-3.24.36/lib/libgtk-3.so.0
No symbol table info available.
#40 0x00007fc7cc977a81 in gtk_event_controller_handle_event () from /nix/store/2dm9a1d865d4s49w8p1lxfwwp7gpz2mp-gtk+3-3.24.36/lib/libgtk-3.so.0
No symbol table info available.
#41 0x00007fc7ccb4188d in _gtk_widget_run_controllers () from /nix/store/2dm9a1d865d4s49w8p1lxfwwp7gpz2mp-gtk+3-3.24.36/lib/libgtk-3.so.0
No symbol table info available.
#42 0x00007fc7cc89da54 in _gtk_marshal_BOOLEAN__BOXEDv () from /nix/store/2dm9a1d865d4s49w8p1lxfwwp7gpz2mp-gtk+3-3.24.36/lib/libgtk-3.so.0
No symbol table info available.
#43 0x00007fc7cd3cf619 in _g_closure_invoke_va (closure=closure@entry=0xc58d70, return_value=return_value@entry=0x7ffe305375a0, instance=instance@entry=0x150b270, args=args@entry=0x7ffe30537670, n_params=1, param_types=0xc58da0) at ../gobject/gclosure.c:895
        marshal = 0x7fc7cd3cdbc0 <g_type_class_meta_marshalv>
        marshal_data = 0x188
        in_marshal = 0
        real_closure = 0xc58d50
        __func__ = "_g_closure_invoke_va"
#44 0x00007fc7cd3e881e in g_signal_emit_valist (instance=0x150b270, signal_id=61, detail=<optimized out>, var_args=var_args@entry=0x7ffe30537670) at ../gobject/gsignal.c:3456
        return_accu = <optimized out>
        accu = {g_type = 20, data = {{v_int = 0, v_uint = 0, v_long = 0, v_ulong = 0, v_int64 = 0, v_uint64 = 0, v_float = 0, v_double = 0, v_pointer = 0x0}, {v_int = 0, v_uint = 0, v_long = 0, v_ulong = 0, v_int64 = 0, v_uint64 = 0, v_float = 0, v_double = 0, v_pointer = 0x0}}}
        accumulator = 0xc59030
        emission = {next = 0x0, instance = 0x150b270, ihint = {signal_id = 61, detail = 0, run_type = (G_SIGNAL_RUN_LAST | G_SIGNAL_ACCUMULATOR_FIRST_RUN)}, state = EMISSION_RUN, chain_type = 21992448}
        instance_type = <optimized out>
        emission_return = {g_type = 20, data = {{v_int = 0, v_uint = 0, v_long = 0, v_ulong = 0, v_int64 = 0, v_uint64 = 0, v_float = 0, v_double = 0, v_pointer = 0x0}, {v_int = 0, v_uint = 0, v_long = 0, v_ulong = 0, v_int64 = 0, v_uint64 = 0, v_float = 0, v_double = 0, v_pointer = 0x0}}}
        rtype = 20
        static_scope = 0
        fastpath_handler = <optimized out>
        closure = <optimized out>
        run_type = <optimized out>
        hlist = <optimized out>
        l = <optimized out>
        fastpath = <optimized out>
        instance_and_params = <optimized out>
        signal_return_type = <optimized out>
        param_values = <optimized out>
        node = <optimized out>
        i = <optimized out>
        n_params = <optimized out>
        __func__ = "g_signal_emit_valist"
#45 0x00007fc7cd3e956f in g_signal_emit (instance=<optimized out>, signal_id=<optimized out>, detail=<optimized out>) at ../gobject/gsignal.c:3606
        var_args = {{gp_offset = 24, fp_offset = 48, overflow_arg_area = 0x7ffe30537750, reg_save_area = 0x7ffe30537690}}
#46 0x00007fc7ccb43214 in gtk_widget_event_internal.part () from /nix/store/2dm9a1d865d4s49w8p1lxfwwp7gpz2mp-gtk+3-3.24.36/lib/libgtk-3.so.0
No symbol table info available.
#47 0x00007fc7cc9f845e in propagate_event () from /nix/store/2dm9a1d865d4s49w8p1lxfwwp7gpz2mp-gtk+3-3.24.36/lib/libgtk-3.so.0
No symbol table info available.
#48 0x00007fc7cc9f9e26 in gtk_main_do_event () from /nix/store/2dm9a1d865d4s49w8p1lxfwwp7gpz2mp-gtk+3-3.24.36/lib/libgtk-3.so.0
No symbol table info available.
#49 0x00007fc7cd0f9705 in _gdk_event_emit () from /nix/store/2dm9a1d865d4s49w8p1lxfwwp7gpz2mp-gtk+3-3.24.36/lib/libgdk-3.so.0
No symbol table info available.
#50 0x00007fc7cd1519d2 in gdk_event_source_dispatch () from /nix/store/2dm9a1d865d4s49w8p1lxfwwp7gpz2mp-gtk+3-3.24.36/lib/libgdk-3.so.0
No symbol table info available.
#51 0x00007fc7cd2d4609 in g_main_dispatch (context=0xc408d0) at ../glib/gmain.c:3444
        dispatch = 0x7fc7cd1519b0 <gdk_event_source_dispatch>
        prev_source = 0x0
        begin_time_nsec = 0
        was_in_call = 0
        user_data = 0x0
        callback = 0x0
        cb_funcs = <optimized out>
        cb_data = <optimized out>
        need_destroy = <optimized out>
        source = 0xc413e0
        current = 0xcb8820
        i = 0
        __func__ = "g_main_dispatch"
#52 g_main_context_dispatch (context=context@entry=0xc408d0) at ../glib/gmain.c:4162
No locals.
#53 0x00007fc7cd2d4898 in g_main_context_iterate (context=context@entry=0xc408d0, block=block@entry=1, dispatch=dispatch@entry=1, self=<optimized out>) at ../glib/gmain.c:4238
        max_priority = 2147483647
        timeout = 133
        some_ready = 1
        nfds = <optimized out>
        allocated_nfds = 5
        fds = 0x1957680
        begin_time_nsec = 0
#54 0x00007fc7cd2d492c in g_main_context_iteration (context=context@entry=0xc408d0, may_block=may_block@entry=1) at ../glib/gmain.c:4303
        retval = <optimized out>
#55 0x00007fc7cd4fce3d in g_application_run (application=0xe76130, argc=argc@entry=1, argv=argv@entry=0x7ffe30537b08) at ../gio/gapplication.c:2571
        arguments = 0xcbd1b0
        status = 0
        context = 0xc408d0
        acquired_context = <optimized out>
        __func__ = "g_application_run"
#56 0x000000000045ccdd in main (argc=<optimized out>, argv=<optimized out>) at ../shell/main.c:68
        application = 0xe76130
```

</details>

So the story probably begins from here:

```gdb
#13 reload_sections (self=self@entry=0x18106d0) at ../panels/keyboard/cc-keyboard-manager.c:581
        path = 0x1b3a360 "/nix/store/qpaiq2ka4mmaps77ffvif8x54cykgwa4-gnome-shell-43.2/share/gnome-control-center/keybindings/50-gnome-shell-launchers.xml"
        dir_path = 0x1b3a2d0 "/nix/store/qpaiq2ka4mmaps77ffvif8x54cykgwa4-gnome-shell-43.2/share/gnome-control-center/keybindings"
```

So I guess, some package (probably gnome-session? not sure) pulls in gnome-shell to `XDG_DATA_DIRS` [So it is loaded here](https://github.com/BuddiesOfBudgie/budgie-control-center/blob/v1.2.0/panels/keyboard/cc-keyboard-manager.c#L559)? :thinking:

Reply:
Yes, `gnome-session` seems to be the one pulling `gnome-shell`.

https://github.com/NixOS/nixpkgs/blob/4bb072f0a8b267613c127684e099a70e1f6ff106/pkgs/desktops/gnome/core/gnome-session/default.nix#L116

That also explains why the "Extensions" app shows up in the menu.

I'm not sure how to fix this, maybe we can add a parameter to `gnome-session`?

Reply:
Assuming as said in https://github.com/NixOS/nixpkgs/commit/3acee13364ebb9bbfeff0478cc4a754955210790 we add this just for some GNOME specific desktop files removing this for Budgie sounds okay to me :thinking:

Reply:
I added a `gnomeShellSupport` parameter to `gnome-session`, it's working now :smile: 

![Captura desde 2023-03-29 21-36-44](https://user-images.githubusercontent.com/62166915/228698741-dbf57a24-7244-438a-bc11-b5923f1f95d8.png)